tools/shared/src/main/scala/org/scalajs/core/tools/corelib/CoreJSLibs.scala

```diff
@@ -99,7 +113,27 @@ object CoreJSLibs {
       else "" // blank line preserves line numbers in source maps
     }
 
-    lines.mkString("", "\n", "\n")
+    val content = lines.mkString("", "\n", "\n")
+
+    outputMode match {
+      case OutputMode.ECMAScript51Global =>
+        content
+
+      case OutputMode.ECMAScript51Isolated =>
+        content
+          .replaceAll("ScalaJS\\.d\\.", "\\$d_")
```

Comment:
I'm probably wrong on the level this "replaceAll" operates, and that's probably minor, but doesn't that mean I can't use the string `"ScalaJS.d"` inside my program?